### PR TITLE
feat: Add value component

### PR DESCRIPTION
**Pull request by FrederikBolding**
Adds the `Value` component, which can be used to show two text values side-by-side where one is highlighted. This component is only allowed as a child of `Row`.

Progresses https://github.com/MetaMask/MetaMask-planning/issues/1686

### CODE REVIEW

**Comment by FrederikBolding** [comment]
@metamaskbot update-pr

File: packages/snaps-sdk/src/jsx/components/Value.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { createSnapComponent } from '../component';
+
+/**
+ * The props of the {@link Value} component.
+ *
+ *
```
Comment (Mrtenz) [inline packages/snaps-sdk/src/jsx/components/Value.ts:6]: Missing docs.

File: packages/snaps-sdk/src/jsx/components/Value.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ */
+export type ValueProps = {
+  value: string;
+  extra: string;
```
Comment (Mrtenz) [inline packages/snaps-sdk/src/jsx/components/Value.ts:10]: Should this be optional?

File: packages/snaps-sdk/src/jsx/validation.test.tsx [hunk 3]
```diff
@@ -513,6 +518,37 @@ describe('DividerStruct', () => {
   });
 });
 
+describe('ValueStruct', () => {
+  it.each([<Value extra="foo" value="bar" />])(
+    'validates a double value element',
```
Comment (Mrtenz) [inline packages/snaps-sdk/src/jsx/validation.test.tsx:523]:
```suggestion
    'validates a value element',
```

File: packages/snaps-sdk/src/jsx/components/Value.ts (hunk 2 above)
Comment (FrederikBolding) [reply packages/snaps-sdk/src/jsx/components/Value.ts:10]: The underlying component isn't, but it probably could be? https://github.com/MetaMask/metamask-extension/blob/develop/ui/components/app/confirm/info/row/value-double.tsx#L15-L16

Comment (Mrtenz) [reply packages/snaps-sdk/src/jsx/components/Value.ts:10]: I guess there's no use case for it at the moment. We can reconsider when we accept different kinds of props or values.

Comment (FrederikBolding) [reply packages/snaps-sdk/src/jsx/components/Value.ts:10]: We can always make it optional later yeah

**Pull request merged**